### PR TITLE
Allow slash in attribute key

### PR DESCRIPTION
**Pull request by chinesedfan**
As discussed in https://github.com/sindresorhus/awesome-lint/issues/54#issuecomment-476026529.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -56,6 +56,14 @@ test('commentMaker(node)', function(t) {
     'marker with boolean attributes'
   )
 
+  node = {type: 'html', value: '<!--foo blala/bar-->'}
+
+  t.deepEqual(
+    marker(node),
+    {name: 'foo', attributes: 'blala/bar', parameters: {'blala/bar': true}, node: node},
+    'marker with boolean attributes(includes slash)'
```
Comment (wooorm) [inline test.js]:
```suggestion
    'marker with boolean attributes (with slash)'
```

**Comment by wooorm** [comment]
@chinesedfan This looks good code-wise, but did you see my comments in the linked post as well?

> Message-control and comment-marker conform to HTML, it's like attributes, which I think makes sense for markdown.

Why should we diverge from HTML?

> Note that we also use colons throughout the ecosystem. To differentiate between the 'source' (in this case awesome) and the ruled.

This is pretty useful, actually. There’s benefits in sticking with the system in place.

**Comment by chinesedfan** [comment]
@wooorm I shared the same idea with you that don't diverge from HTML, but confused when you mentioned colons. Did you suggest we use a simple ruleId without slash? Because `source` can contain namespace information? Then it goes back to my advice about renaming.

**Comment by wooorm** [comment]
Again, the code looks good, but I’d like to understand why you think this is a good idea a bit better first, so here goes.

I haven‘t delved into the awesome linter recently, but I believe it’s presenting itself to users as its own thing. Through something like `remark-lint:awesome/badge`. Why not go with `awesome:badge`? Or otherwise, maybe `remark-lint:awesome:badge`?

Why do we need to mimic the file-system by using slashes. Why follow ESLint? What’s wrong with colons?

**Comment by chinesedfan** [comment]
> I shared the same idea with you that don't diverge from HTML.

@wooorm Yes, we are in the same team if re-designing from the beginning. But I can also accept allowing slash in attribute key. The reason why I send this PR is awesome-linter has been already there. In https://github.com/sindresorhus/awesome-lint/issues/54, @sindresorhus didn't want to rename ruleIds, and I didn't recognize your opposition clearly.

So, @sindresorhus, please make an agreement with @wooorm then we can know how to solve it. You are facing "1 vs 2" I think. :P

**Comment by wooorm** [comment]
Seems awesome-lint [can switch to use colons](https://github.com/sindresorhus/awesome-lint/issues/54#issuecomment-485316415), so I’ll close this here.

**Comment by wooorm** [comment]
P.S. thanks for creating the PR tho, @chinesedfan!

**Pull request closed without merging**